shell-control/agent/backend/tools_base.py

```
@@ -0,0 +1,70 @@
+# A decorator for marking methods as tools
+def tool(name: str, formatter_function: Optional[str] = None) -> Callable:
+    def decorator(func: Callable) -> Callable:
+        func._is_tool = True  # Mark the function as a tool
+        func._tool_name = name  # Set the name of the tool
+        func._formatter_function = formatter_function  # Optional formatter function
+        return func
+    return decorator
+
+
+class ToolsBase:
+    def __init__(self, settings: dict = {}):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)